### ISSUE TITLE
Action Required: Duplicate Tab Shortcut Requires Changes to Comply with Chrome Web Store Policy

### ISSUE THREAD

**Issue by stefansundin** [stefansundin/duplicate-tab#10]
Received this email today:

> Dear Developer,
> 
> Your Google Chrome item "Duplicate Tab Shortcut" with ID: klehggjefofgiajjfpoebdidnpjmljhb did not comply with our policies.
> 
> Your item is still published, but is at risk of being removed from the Chrome Web Store.
> 
> 
> Your item did not comply with the following section of our [Program Policies](https://developer.chrome.com/webstore/program_policies):
> 
> 
> "User Data Privacy"
> 
> Your product violates the "Use of Permissions" section of the policy, which requires that you:
> 
> - Request access to the narrowest permissions necessary to implement your product’s features or services.
> - If more than one permission could be used to implement a feature, you must request those with the least access to data or functionality.
> - Don't attempt to "future proof" your product by requesting a permission that might benefit services or features that have not yet been implemented.
> 
> Please make the necessary changes within 14 days in order to avoid removal.

As always, Google has machines write these emails, so it's never clear what they want. I replied asking for clarification, and hopefully they can give some.

cc @bbarenblat. Your co-workers are trying to remove my extension.

**Comment by bbarenblat** [comment]
Ugh, I’m sorry. I’ll poke around internally and see if I can find the people responsible for herding the robots that do these extension culls. In the meantime, my guess is that the robots are unhappy about the `tabs` permission, even though it’s an opt-in. I seem to remember that `activeTab` was insufficient to open the new tab in the background – is that still the case?

**Comment by stefansundin** [comment]
I have not done any test recently if the way the permissions work has changed. I really doubt it since it needs to read [`tab.url`](https://github.com/stefansundin/duplicate-tab/blob/2442aff6a99bc3661045e05bd55d32bf1e686ba5/chrome/background.js#L43).

Still hasn't gotten any reply from a human, but I did get this email:

> Dear Developer, 
> 
> 
> Upon review of your Product, we find that it does not comply with the Chrome Web Store’s User Data Policy, and it has been removed from the store.
> 
> 
> Your Product violates the “Use of Permissions” section of the policy, which requires that you:
> 
> 
> Request access to the narrowest permissions necessary to implement your Product’s features or services. If more than one permission could be used to implement a feature, you must request those with the least access to data or functionality.
> 
> 
> Don't attempt to "future proof" your Product by requesting a permission that might benefit services or features that have not yet been implemented.
> 
> 
> To reinstate your Product, please ensure that your Product requests and uses only those permissions that are necessary to deliver the currently stated product’s features. 
> 
> 
> If you’d like to re-submit your Product, please modify your Product so that it complies with the Chrome Web Store’s Developer Program Policies, then re-publish it in your Developer Dashboard. 
> 
> 
> Please reply to this email for questions / clarifications regarding this Product removal.
> 
> 
> Thank you for your cooperation,
> 
> Google Chrome Web Store team

However, I can still see it published, so not quite sure what is going on. Maybe a human reverted it? Very frustrating that I'm just on the receiving end of automated emails.

If it ends up being removed, I was thinking that I may just remove everything that uses the tabs permission. Basically the extension would be reduced to only the first shortcut. It's what I mostly use anyway. But it would mean that optional permissions are entirely useless.

**Comment by stefansundin** [comment]
I never received a reply to my email to try to get a clarification. And now they sent this.. WTF?

> Dear Developer,
> 
> Your Google Chrome item "Duplicate Tab Shortcut" with ID: klehggjefofgiajjfpoebdidnpjmljhb did not comply with our policies and was removed from the Chrome Web Store.
> 
> We have not received an update from you on your Google Chrome item before the expiration of the warning period specified in our earlier email.
> 
> Because your item continues to not comply with our policies stated in the previous email, it has now been removed from the Chrome Web Store.
> 
> Once your item complies with Chrome Web Store policies, you may request re-publication in the Chrome Web Store Developer Dashboard. Your item will be reviewed for policy compliance prior to re-publication.
> 
> If you have any questions about this email, please respond and the Chrome Web Store Developer Support team will follow up with you.

Just to be clear, I *did* reply to their original email. I will reply again to this one. Unbelievable.

**Comment by bbarenblat** [comment]
Hey – I meant to reply on this yesterday. I found the appropriate people at work, and this has been escalated to get a second opinion. The extension is still in the store, so I think this latest message may just be a disconnect between the robot that scolds people over email and the robot that actually does deletions. I’ll let you know if I hear any more.

**Comment by stefansundin** [comment]
Hi @bbarenblat. Thanks for your help.

I am still getting a 404 when trying to access https://chrome.google.com/webstore/detail/duplicate-tab-shortcut-ke/pfippblampohahkkdoomekekmfbjkimg.

I did get a reply last night, which said:

> Dear Developer,
> 
> Thank you for reaching out to us. Upon a subsequent review, we’ve reinstated your item and it will be available in the Chrome Web Store soon.
> 
> Thank you for your cooperation,
> 
> Chrome Web Store team

So maybe it is just a matter of time, but it's been 10+ hours since I got this email so this seems like a long time.

**Comment by maiermic** [comment]
I was wondering, why the shortcut did not work anymore today. The extension got disabled due to

> This extension violates the Chrome Web Store policy.

@stefansundin Did you get another mail?

**Comment by stefansundin** [comment]
Hey @maiermic. Nope, I haven't received anything. Looks to me like the extension is still available. Maybe they accidentally did something and then reverted it a minute later?

Can you check again? If it's still blocked, can you provide a screenshot that includes the URL? Thanks!

**Comment by maiermic** [comment]
I can enable it, but the warning is still shown in chrome://extensions/
![screenshot-duplicate-tab-shortcut](https://user-images.githubusercontent.com/8530117/134685869-6910d754-0397-471b-9c09-1a0c4d1856be.png)

The [details page](chrome://extensions/?id=pfippblampohahkkdoomekekmfbjkimg) looks like this 
![screenshot-duplicate-tab-shortcut_details](https://user-images.githubusercontent.com/8530117/134686185-1de0319d-3a62-4f19-86a2-d3d0088cde1e.png)

**Comment by stefansundin** [comment]
So that's a different extension. This one is mine: https://chrome.google.com/webstore/detail/duplicate-tab-shortcut/klehggjefofgiajjfpoebdidnpjmljhb

😅

**Comment by maiermic** [comment]
My fault :sweat_smile: 

> Most likely it was removed due to the "Read your browsing history" permission, and the author didn't fix it in time after Google notified him.

You're probably right about that. Too bad that the (information about the) extension is not available anymore in the Chrome Web Store. How should you contact/find out the author of an extension? :disappointed: 

Thanks for your response and I consider your extension as an alternative :wink:

**Comment by stefansundin** [comment]
> You're probably right about that. Too bad that the (information about the) extension is not available anymore in the Chrome Web Store. How should you contact/find out the author of an extension? 😞

No idea. Nowadays Google requires all extension authors to have a public email address on the extension page (this was not required previously). But if that page is inaccessible then it may be hard. If you still have it installed then you may be able to find the extension code somewhere on your hard drive, and if you're lucky then the author put it on GitHub and you may be able to search for certain parts of the code to find it.

> Thanks for your response and I consider your extension as an alternative 😉

I never tried the extension that you used. Did it do anything well that this extension is missing?

If you're able to find the code as I suggested above, then you can load the code in developer mode and keep using the other extension.

**Comment by maiermic** [comment]
> I never tried the extension that you used. Did it do anything well that this extension is missing?

I'm not sure if the extension just existed before yours or if I tried yours in the past and was missing something. I remember that one extension that I tried back then did not preserve the navigation history, which was/is important to me, but it seems that yours supports this feature.

In the meantime, I found out that this feature has been added to another extension that I'm already using: [Quick Tabs](https://chrome.google.com/webstore/detail/quick-tabs/jnjfeinjfmenlddahdjdmgpbokiacbbb). Hence, I don't need an extra extension.
